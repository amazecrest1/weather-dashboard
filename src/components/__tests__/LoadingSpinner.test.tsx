import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import LoadingSpinner from '../LoadingSpinner';

describe('LoadingSpinner', () => {
  it('renders loading spinner', () => {
    const { container } = render(<LoadingSpinner />);
    
    const spinner = container.querySelector('.animate-spin');
    expect(spinner).toBeInTheDocument();
  });

  it('applies default medium size', () => {
    const { container } = render(<LoadingSpinner />);
    
    const spinner = container.querySelector('.animate-spin');
    expect(spinner).toHaveClass('h-8', 'w-8');
  });

  it('applies small size when specified', () => {
    const { container } = render(<LoadingSpinner size="sm" />);
    
    const spinner = container.querySelector('.animate-spin');
    expect(spinner).toHaveClass('h-4', 'w-4');
  });

  it('applies large size when specified', () => {
    const { container } = render(<LoadingSpinner size="lg" />);
    
    const spinner = container.querySelector('.animate-spin');
    expect(spinner).toHaveClass('h-12', 'w-12');
  });

  it('applies custom className', () => {
    const { container } = render(<LoadingSpinner className="custom-class" />);
    
    const spinnerContainer = container.querySelector('.flex');
    expect(spinnerContainer).toHaveClass('custom-class');
  });

  it('has proper styling classes', () => {
    const { container } = render(<LoadingSpinner />);
    
    const spinnerContainer = container.querySelector('.flex');
    expect(spinnerContainer).toHaveClass('flex', 'items-center', 'justify-center');
    
    const spinner = container.querySelector('.animate-spin');
    expect(spinner).toHaveClass(
      'animate-spin',
      'rounded-full',
      'border-b-2',
      'border-primary-600'
    );
  });

  it('combines custom className with default classes', () => {
    const { container } = render(<LoadingSpinner className="mt-4 custom-spacing" />);
    
    const spinnerContainer = container.querySelector('.flex');
    expect(spinnerContainer).toHaveClass('flex', 'items-center', 'justify-center');
    expect(spinnerContainer).toHaveClass('mt-4', 'custom-spacing');
  });

  it('matches snapshot with default size', () => {
    const { container } = render(<LoadingSpinner />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with small size', () => {
    const { container } = render(<LoadingSpinner size="sm" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with large size', () => {
    const { container } = render(<LoadingSpinner size="lg" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with custom className', () => {
    const { container } = render(<LoadingSpinner className="custom-class" />);
    expect(container.firstChild).toMatchSnapshot();
  });
}); 