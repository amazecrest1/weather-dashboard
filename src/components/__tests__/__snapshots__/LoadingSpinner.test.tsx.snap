// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner matches snapshot with custom className 1`] = `
<div
  class="flex items-center justify-center custom-class"
>
  <div
    class="animate-spin rounded-full border-b-2 border-blue-600 dark:border-blue-400 h-8 w-8"
  />
</div>
`;

exports[`LoadingSpinner matches snapshot with default size 1`] = `
<div
  class="flex items-center justify-center "
>
  <div
    class="animate-spin rounded-full border-b-2 border-blue-600 dark:border-blue-400 h-8 w-8"
  />
</div>
`;

exports[`LoadingSpinner matches snapshot with large size 1`] = `
<div
  class="flex items-center justify-center "
>
  <div
    class="animate-spin rounded-full border-b-2 border-blue-600 dark:border-blue-400 h-12 w-12"
  />
</div>
`;

exports[`LoadingSpinner matches snapshot with small size 1`] = `
<div
  class="flex items-center justify-center "
>
  <div
    class="animate-spin rounded-full border-b-2 border-blue-600 dark:border-blue-400 h-4 w-4"
  />
</div>
`;
